src/models/student.py

```diff
@@ -11,6 +12,87 @@ class Student(db.Model):
     name = db.Column(db.String(200))
     created_at = db.Column(db.DateTime, default=datetime.datetime.utcnow)
 
+    @staticmethod
+    def generate_enrolment_key(mobile, from_helpline):
+        '''
+            this function helps to create a a record of a new student in the database were the from_helpine varibale helps to track was the call from helpline or was manually created and send the enrollment key to the mobile number
+
+            if was from_helpline record the incoming calls also
+            
+            USAGE :
+                Student.generate_enrolment_key(mobile, from_helpline)
+
+            params: 
+                mobile : String required
+                from_helpline : Boolean required
+
+        '''
+        student = Student(stage=app.config['STAGES']['EKG'])
+        db.session.add(student)
+        db.session.commit()
+        student_contact = StudentContact(contact=mobile, main_contact=False, student_id=student.id)
+        db.session.add(student_contact)
+        db.session.commit()
+
+        #Send sms when the call is from helpine else don't send the sms
+        message = student.send_enrolment_key(from_helpline)
+
+        return message
+
+    def send_enrolment_key(self, from_helpline):
+        '''
+            Method is used to send valid enrollment key to the student if exist
+            else it will generate a new enrollment key and send it to the user
+
+            USAGE: 
+                instance.send_enrollment_key(from_helpline)
+
+            params:
+                from_helpline : Boolean required
+
+        '''
+        student_id = self.id
+        
+        enrollment = EnrolmentKey.query.filter_by(student_id=student_id).order_by(EnrolmentKey.created_at.desc()).first()
+
+        student_contact = StudentContact.query.filter_by(student_id=student_id,main_contact=False).order_by(StudentContact.created_at.desc()).first()
```

Comment:
Make sure we are sending the Enrolment key to all student contacts and not just a single one.